File: Flutter/webadview_flutter/android/src/main/kotlin/dk/stepnetwork/webadview_flutter/core/AdLoadState.kt
package dk.stepnetwork.webadview_flutter.core

/**
 * Lazy-loading lifecycle state of an ad unit (port of AdLoadState).
 * notLoaded → fetched (fetch zone) → displayed (display zone) → unloaded
 * (outside the unload zone for the stability delay, only with unloading
 * enabled) → notLoaded on re-entry. [wireName] is the Dart contract string.
 */
enum class AdLoadState(val wireName: String) {
    NOT_LOADED("notLoaded"),
    FETCHED("fetched"),
    DISPLAYED("displayed"),
    UNLOADED("unloaded"),
}

/** Tuning values for lazy loading (port of LazyLoadingConfig). Distances in dp. */
data class LazyLoadingConfig(
    val fetchThreshold: Double = 800.0,
    val displayThreshold: Double = 200.0,
    val unloadThreshold: Double = 1600.0,
    val unloadingEnabled: Boolean = false,
)

File: Flutter/webadview_flutter/lib/src/model/ad_load_state.dart
/// Lazy-loading lifecycle state of an ad unit (mirrors the SDK's
/// `AdLoadState`).
///
/// `notLoaded → fetched` (the ad page loads) `→ displayed` (the creative
/// renders) `→ unloaded` (far away for 2 s, only with unloading enabled)
/// `→ notLoaded` on re-entry.
enum AdLoadState {
  notLoaded,
  fetched,
  displayed,
  unloaded;

  /// Wire name on the method channel (equals the Swift raw value).
  String get wireName => name;

  /// Parses a wire name; `null` for unknown values (never throws — the
  /// value comes from the platform side).
  static AdLoadState? tryParse(Object? value) {
    if (value is! String) return null;
    for (final state in values) {
      if (state.name == value) return state;
    }
    return null;
  }
}
